src/cassiopeia/tools/parameter_estimators.py

```diff
@@ -1,170 +1,219 @@
+        characters_key = kwargs.pop("layer")
 
-    if character_matrix is None:
-        raise ParameterEstimateError("No character matrix is detected in this tree.")
+    character_matrix = utils._get_character_matrix(tree, characters_key)
+    missing_state_indicator = utils._get_missing_state_indicator(tree, missing_state)
 
-    num_dropped = (character_matrix.values == tree.missing_state_indicator).sum()
+    num_dropped = utils._count_entries(character_matrix, missing_state_indicator)
     missing_proportion = num_dropped / (character_matrix.shape[0] * character_matrix.shape[1])
     return missing_proportion
 
 
-def get_proportion_of_mutation(tree: CassiopeiaTree, layer: str | None = None) -> float:
-    """Calculates the proportion of mutated entries in the character matrix.
+def get_proportion_of_mutation(
+    tree: CassiopeiaTree | TreeData,
+    characters_key: str = "characters",
+    missing_state=None,
+    unmodified_state=None,
+    **kwargs,
+) -> float:
+    """Calculate the proportion of mutated entries in the character matrix.
 
     Calculates the proportion of cell/character entries in the character matrix
     that have a non-uncut (non-0) state, normalizing over non-missing entries.
     Hence, missing entries are not considered in calculating the proportion.
 
     Args:
-        tree: The CassiopeiaTree specifying the tree and the character matrix
-        layer: Layer to use for character matrix. If this is None,
-            then the current `character_matrix` variable will be used.
+        tree: CassiopeiaTree or TreeData object containing the character matrix
+        characters_key: Key for the character matrix. For CassiopeiaTree, if "characters",
+            uses the default character_matrix attribute; otherwise looks in layers.
+            For TreeData, specifies the obsm key. Default is "characters".
+        missing_state: Value(s) to consider as missing data. Can be a single value
+            or a list of values. If None, uses the tree's missing_state_indicator
+            attribute, or defaults to [-1, "-1", "NA", "-"]. Default is None.
+        unmodified_state: Value(s) to consider as unmodified/uncut states. Can be
+            a single value or a list of values. If None, defaults to [0, "0", "*"]
+            for flexibility with both integer and string character matrices.
+            Default is None.
+        **kwargs: Deprecated arguments. Use 'characters_key' instead of 'layer'.
 
     Returns:
-            The proportion of non-missing cell/character entries that are mutated
+        Proportion of non-missing cell/character entries that are mutated (between 0 and 1)
 
     Raises:
-            ParameterEstimateError if character matrix or layer doesn't exist
+        ParameterEstimateError: If character matrix or layer doesn't exist
     """
-    if layer:
-        character_matrix = tree.layers[layer]
-    else:
-        character_matrix = tree.character_matrix
+    if "layer" in kwargs:
```

Comment:
Try to avoid duplicated code. Maybe create a helper function or add to _get_character_matrix?

```
character_matrix = utils._get_character_matrix(tree, characters_key,**kwargs)
```